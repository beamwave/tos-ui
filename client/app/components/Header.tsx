import React, { Component, createRef } from 'react'
import { Link, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap'
import { throttle, debounce } from 'throttle-debounce'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import {
  sidebar,
  toggleEditMode,
  changeCategory,
  trainingWheelsProtocol
} from '../actions/app'
import { query, startSearch } from '../actions/filter'
import { startLogout } from '../actions/auth'
import { loadModal } from '../actions/modal'
import { ADD_IMAGE_MODAL } from '../constants/modaltypes'

interface RProps {
  photo?: string
  username?: string
  isAuthenticated?: boolean
  trainingWheels?: boolean
  query?: any
  input?: string
  category?: string
  backButton?: boolean
  history?: any
}

interface RState {
  sidebar?: any
  search?: any
  startSearch?: (query: any) => any
  toggleEditMode?: () => any
  trainingWheelsProtocol?: () => any
  changeCategory?: (string) => any
  loadModal?: (string) => any
  startLogout?: () => void
}

type Props = RProps & RState

export class Header extends Component<Props> {
  state = {
    // data: {
    //   query: '',
    //   category: 'art'
    //   // category: this.props.category && this.props.category
    // },
    input: '',
    dropdownOpen: false
    // showSearch: false,
    // editMode: false
  }

  toggleDropdown = () =>
    this.setState({ dropdownOpen: !this.state.dropdownOpen })

  toggleTrainingWheels = () => this.props.trainingWheelsProtocol()

  toggleEditMode = () => this.props.toggleEditMode()

  // needs to be attached to document for click detection
  // componentDidMount = () => {
  //   const input = this.inputRef.current
  //   if (input) input.focus()

  //   document.addEventListener('click', this.determineClick, false)
  // }

  // needs to be attached to document for click detection
  // componentWillUnmount = () =>
  //   document.removeEventListener('click', this.determineClick)

  // addImageModal = () => this.props.loadModal(ADD_IMAGE_MODAL)

  // handleCategoryChange = ({ target }) => {
  //   this.props.changeCategory(target.value)
  // }

  // determineClick = e => {
  //   // if click outside modal or not typing in input, hide search
  //   if (!e.target.closest('.section') && !e.target.dataset.search) {
  //     this.hideSearch()
  //   }

  //   // else if you click inside or type in input, ...
  // }

  // showSearch = () => this.setState({ showSearch: true })

  // hideSearch = () => this.setState({ showSearch: false })

  // private handleCall = async data => this.props.startSearch(data)

  // // should debounce action creator in handleCall for 5 seconds
  // private startSearchDebounce = debounce(1000, this.handleCall)
  // private startSearchThrottle = throttle(750, this.handleCall)

  // private handleText = input => this.setState({ input })

  // private handleSearch = async e => {
  //   e.preventDefault()
  //   // value from onChange
  //   const {
  //     target: { value: input }
  //   } = e
  //   const { category, startSearch } = this.props

  //   this.handleText(input)

  //   // data sent to action creator
  //   const data = {
  //     input,
  //     tags: input.replace(/\s/g, '').split(','),
  //     category: category
  //   }

  //   if (input.length < 5) {
  //     this.startSearchThrottle(data)
  //   } else {
  //     this.startSearchDebounce(data)
  //   }
  // }

  // handleSelectColor = category => {
  //   if (category === 'art') return 'select teal'
  //   if (category === 'comments') return 'select violet'
  //   if (category === 'clothes') return 'select blue'
  //   if (category === 'people') return 'select grey'
  // }

  private inputRef = React.createRef<HTMLInputElement>()

  render = () => {
    const {
      username,
      isAuthenticated,
      startLogout,
      history
      // photo,
      // trainingWheels,
      // backButton,
    } = this.props
    // const {
    //   input,
    //   data: { category }
    // } = this.state

    return (
      <header className="global-header">
        <p className="logo">Collegeboard</p>
        <div className="user">
          <p>{username}</p>
          <div className="profile-image" />
        </div>
      </header>
      // <header className="nav-header">
      //   {!backButton ? (
      //     <div className="plus" onClick={this.addImageModal}>
      //       <FontAwesomeIcon icon="plus" />
      //       <p className="hint">upload image</p>
      //     </div>
      //   ) : (
      //     <div className="plus" onClick={() => history.goBack()}>
      //       <FontAwesomeIcon icon="arrow-left" />
      //       <p className="hint">go back</p>
      //     </div>
      //   )}
      //   <div className="settings">
      //     <FontAwesomeIcon icon="cog" />
      //     <p className="hint">settings</p>
      //   </div>
      //   {!backButton && (
      //     <div className="protocol" onClick={this.toggleTrainingWheels}>
      //       {trainingWheels ? (
      //         <FontAwesomeIcon icon="lock" />
      //       ) : (
      //         <FontAwesomeIcon icon="unlock" />
      //       )}
      //       <p className="hint">
      //         {trainingWheels ? 'deactive' : 'active'} training wheels protocol
      //         )}
      //       </p>
      //     </div>
      //   )}
      //   {backButton && (
      //     <div className="protocol" onClick={this.toggleEditMode}>
      //       <FontAwesomeIcon icon="sliders-h" />
      //       <p className="hint">adjust image settings</p>
      //     </div>
      //   )}
      //   <div className="search-group">
      //     <FontAwesomeIcon icon="search" className="icon" />
      //     <form className="submit" onSubmit={this.handleSearch}>
      //       <input
      //         type="text"
      //         className="input spawnSearch"
      //         placeholder="Search"
      //         onChange={this.handleSearch}
      //         value={input}
      //         ref={this.inputRef}
      //       />

      //       <div className="type-choice">
      //         <div className="angle-down">
      //           <FontAwesomeIcon icon="angle-down" className="icon" />
      //         </div>
      //         <select
      //           className={this.handleSelectColor(category)}
      //           value={category}
      //           onChange={this.handleCategoryChange}
      //         >
      //           <option value="art">art</option>
      //           <option value="comments">comments</option>
      //           <option value="clothes">clothes</option>
      //           <option value="people">people</option>
      //           <option value="music">music</option>
      //           <option value="quora">quora</option>
      //           <option value="bookmarks">bookmarks</option>
      //           <option value="recipes">recipes</option>
      //           <option value="vocabulary">vocabulary</option>
      //           <option value="notes">notes</option>
      //           <option value="timeline">timeline</option>
      //         </select>
      //       </div>
      //       <div className="search-options">
      //         <p className="save-search">save search</p>
      //         <p className="advanced-search">advanced search</p>
      //       </div>
      //     </form>
      //   </div>

      //   <Dropdown
      //     isOpen={this.state.dropdownOpen}
      //     toggle={this.toggleDropdown}
      //     className="dropdown-root"
      //   >
      //     <DropdownToggle className="dropdown-toggle">
      //       <p className="nav-username">{username}</p>
      //       <div
      //         className="image"
      //         style={{
      //           // url bg-pos: x% y% / bg-scale: %
      //           background: `url(${photo}) 50% 5% / 110% no-repeat`,
      //           marginRight: 15
      //         }}
      //       />
      //       <FontAwesomeIcon icon="angle-down" className="icon fa-angle-down" />
      //     </DropdownToggle>
      //     <DropdownMenu
      //       right
      //       className="dropdown-menu"
      //       style={{
      //         display: this.state.dropdownOpen === false ? 'none' : 'block'
      //       }}
      //     >
      //       <DropdownItem className="dropdown-item">
      //         <Link to="/profile" className="profile">
      //           Profile
      //         </Link>
      //       </DropdownItem>
      //       <DropdownItem
      //         className="dropdown-item"
      //         onClick={() => startLogout()}
      //       >
      //         {isAuthenticated ? <p className="logout">Logout</p> : null}
      //       </DropdownItem>
      //     </DropdownMenu>
      //   </Dropdown>
      // </header>
    )
  }
}

const mapStateToProps = (state, props) => ({
  // photo: state.auth.profileImage,
  username: state.auth.username
  // isAuthenticated: !!state.auth.token,
  // category: state.app.category,
  // trainingWheels: state.app.trainingWheels,
  // backButton: props.match.params._id ? true : false // for some reason this works
})

export default withRouter<any>(
  connect<RProps, any>(
    mapStateToProps,
    {
      sidebar,
      query,
      startSearch,
      toggleEditMode,
      changeCategory,
      trainingWheelsProtocol,
      loadModal,
      startLogout
    }
  )(Header)
)
