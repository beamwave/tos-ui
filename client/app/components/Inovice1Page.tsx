import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { loadModal } from '../actions/modal'
import { GRADE_ADJUSTMENTS_MODAL } from '../constants/modaltypes'

import moment from 'moment'
import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

import { startTestTakerRetrieval } from '../actions/testTaker'

interface IProps {
  results: any
  loadModal: any
  testTakerData: any
  startTestTakerRetrieval: (data: any) => any
}

export class Invoice1Page extends Component<IProps> {
  state = {}

  gradeAdjustmentModal = () => this.props.loadModal(GRADE_ADJUSTMENTS_MODAL)

  componentDidMount = () => {
    const { startTestTakerRetrieval } = this.props
    // const { getDataFromStore } = this.state
    const data = { aicode: '471315' }
    startTestTakerRetrieval(data)
  }

  render = () => {
    const { results, testTakerData } = this.props

    console.log('test taker data:', testTakerData)

    return (
      <div className="invoice-1-page-container">
        <header>
          <div className="progress-bar">
            <div className="titles">
              <p>School Info</p>
              <p>Borrowed/Loaned Test Books</p>
              <p>Confirmation</p>
            </div>
            <div className="stages">
              <div className="stage" />
              <div className="stage" />
              <div className="stage" />
              <div className="line" />
            </div>
          </div>
        </header>
        <main>
          <section className="body">
            <section>
              <h2>Provide information to help us calculate your invoice</h2>
              <p>
                The information you are about to provide is used to calculate
                your invoice. Invoices will be sent to schools in mid-January.
                You must provide invoice information by 1/11/2019. You cannot
                make edits after this date.
              </p>
            </section>
            <section>
              <h2>Send bill to</h2>
              <form className="user-information-form">
                <div className="form-group">
                  <label>First Name *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Last Name *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Job Title *</label>
                  <select>
                    <option value="select">- Select Job Title -</option>
                    <option value="c">K12: Counselor</option>
                    <option value="tc">K12: Test Coordinator</option>
                    <option value="apc">K12: AP Coordinator</option>
                    <option value="p">K12: Principal</option>
                    <option value="ap/d">K12: Assistant Principal/Dean</option>
                    <option value="dh">K12: Department Head</option>
                    <option value="sf">K12: School Faculty</option>
                    <option value="apt">K12: AP Teacher</option>
                    <option value="t">K12: Teacher (non-AP)</option>
                    <option value="cs">K12: Curriculum Specialist</option>
                    <option value="o">K12: Other - K-12</option>
                  </select>
                </div>
                <div className="form-group">
                  <label>Email address *</label>
                  <input />
                </div>
                <div className="form-group">
                  <label>Purchase Order Number</label>
                  <input />
                </div>
              </form>
            </section>
            <section>
              <h2>Students tested</h2>
              <p>
                The numbers below show the number of standard tests shipped (for
                your reference) and the number of student answer sheets received
                and will be used to calculate your bill. If this isn't what you
                are expecting, it is possible that a student may have indicated
                an incorrect grade. Grade adjustments may be made for a limited
                time to ensure that your bill is accurate. Please note that any
                adjustments made here will not be reflected in score reports.
              </p>
              <div className="grade-adjustments">
                <p
                  className="link grade-adjustments-button"
                  onClick={this.gradeAdjustmentModal}
                >
                  Grade Adjustments
                </p>
                <p className="grade-adjustments-tooltip">
                  Grade is often used to determine who to bill for student test
                  fees (for example, whether fees are billed to the school,
                  district, or state), and occasionally students may indicate an
                  incorrect grade. Grade adjustments may be made for a limited
                  time to ensure that your bill is accurate. Please note that
                  any adjustments made here will not be reflected in score
                  reports. Also note that grade adjustments will be discarded if
                  there is a subsequent update to your school's answer sheet
                  data; any previous grade adjustments will need to be
                  re-entered.
                </p>
              </div>

              <div className="students-tested-table">
                <div className="headers">
                  <p>Grade</p>
                  <p>Tests Shipped</p>
                  <p>Ans Sheets Received</p>
                  <p>Contract</p>
                </div>
                {testTakerData.map(
                  ({ grade, shipped, ansSheets, contract }) => (
                    <div
                      className="grade-row"
                      key={`${grade}-${shipped}-${ansSheets}`}
                    >
                      <p>{grade}</p>
                      <p>{shipped}</p>
                      <p>{ansSheets}</p>
                      <p>{contract}</p>
                    </div>
                  )
                )}
                <div className="total-row">
                  <p>Total</p>
                  <p>
                    {testTakerData.reduce(
                      (store, { shipped }) => store + shipped,
                      0
                    )}
                  </p>
                  <p>
                    {testTakerData.reduce(
                      (store, { ansSheets }) => store + ansSheets,
                      0
                    )}
                  </p>
                </div>
              </div>
            </section>
            <section>
              <h2>Students covered by contracts</h2>
              <p>(table varies between psat/sat/)</p>
              <div className="contract-table">
                <div className="headers">
                  <p>Covered by</p>
                  <p>SAT (witout Essay)</p>
                  <p>SAT with Essay</p>
                </div>
                <div className="state">
                  <p>State Contract</p>
                  <p>0</p>
                  <p>0</p>
                </div>
                <div className="district">
                  <p>District Contract</p>
                  <p>0</p>
                  <p>0</p>
                </div>
              </div>
            </section>
          </section>
          <aside>
            <div className="sidepanel-container">
              {/* <h3>{name}</h3>
            <p>{aicode}</p>
            <p>{admin}</p>
            <p>{date}</p> */}
              <h3>Princess Anne high school</h3>
              <p>(471315)</p>
              <p>Sat School Day</p>
              <p>Wed. October 10, 2018</p>
              <p>Reason for modifying or submitting invoice</p>
              <textarea />
              <Link to="/invoice" className="blue-link-button">
                Continue
              </Link>
              <Link to="/school-landing" className="link">
                exit
              </Link>
            </div>
          </aside>
        </main>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  results: state.school.results[0], // change to know which school was clicked on
  testTakerData: state.testTaker.data
})

export default connect(
  mapStateToProps,
  { startTestTakerRetrieval, loadModal }
)(Invoice1Page)
