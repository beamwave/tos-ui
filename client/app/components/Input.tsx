import React, { Component } from 'react'

const Input = ({ query, handleChange }) => {
  const Input = (
    <input
      type="text"
      className="input spawnSearch"
      placeholder="Search"
      onChange={handleChange}
      value={query}
    />
  )

  return Input
}

export default Input

// import React, { Component } from 'react'

// interface IProps {
//   handleSearch: (e: any) => Promise<void>
// }

// export class Input extends Component<IProps> {
//   state = {
//     query: ''
//   }

//   handleChange = e => {
//     e.preventDefault()
//     const {
//       target: { value: query }
//     } = e
//     this.setState({ query })
//     this.props.handleSearch(e)
//   }

//   shouldComponentUpdate = (nextProps, nextState) => {
//     console.log('next state', this.state.query)
//     console.log('incoming query', nextState.query)

//     if (nextState.query !== this.state.query) return true

//     return false
//   }
//   render = () => {
//     const { query } = this.state
//     return (
//       <input
//         type="text"
//         className="input spawnSearch"
//         placeholder="Search"
//         onChange={this.handleChange}
//         value={query}
//       />
//     )
//   }
// }

// export default Input
