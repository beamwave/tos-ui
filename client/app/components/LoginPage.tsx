import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '../../../node_modules/@fortawesome/react-fontawesome'

import { startLogin } from '../actions/auth'

interface ICredentials {
  username: string
  password: string
}

interface IProps {
  history?: any
  startLogin(data: ICredentials): any
}

interface IState {
  username: string
  password: string
  errors: {
    username: string
    password: string
    global?: string
  }
}

export class LoginPage extends Component<IProps, IState> {
  state = {
    username: '',
    password: '',
    errors: {
      username: '',
      password: '',
      global: ''
    }
  }

  handleCredentialError = property =>
    this.setState(prevState => ({
      errors: {
        ...prevState.errors,
        [property]: `${property} password field cannot be blank`
      }
    }))

  handleCredentialReset = property =>
    this.setState(prevState => ({
      errors: {
        ...prevState.errors,
        [property]: ``
      }
    }))

  onFieldChange = ({ target }) => {
    const { name, value }: { name: keyof IState; value: string } = target
    this.setState({ [name]: value } as any)
  }

  onSubmit = async e => {
    e.preventDefault()

    // extract submission helpers from props
    const { startLogin, history } = this.props

    // extract entered username and password from state
    const { username, password } = this.state

    // build credentials object for submission
    const credentials = { username, password }

    // check that username is not empty
    if (username.length === 0) this.handleCredentialError(`username`)
    if (username.length !== 0) this.handleCredentialReset(`username`)

    // check that password is not empty
    if (password.length === 0) this.handleCredentialError(`password`)
    if (password.length !== 0) this.handleCredentialReset(`password`)

    // if frontend validations pass, send user credentials to the server
    if (username.length > 0 && password.length > 0) {
      try {
        await startLogin(credentials)
        console.log('login success')
        history.push('/dashboard')
      } catch (err) {
        console.log('frontend error', err)
        return this.setState({ errors: err.response.data.errors })
      }
    }
  }

  render = () => {
    const { errors } = this.state
    return (
      <div className="credentials-container">
        <section className="login-section">
          <h1>Welcome back. Please sign in.</h1>
          <hr />
          <form onChange={this.onFieldChange} onSubmit={this.onSubmit}>
            <hr />
            <label>Username</label>
            <input name="username" />
            <label>Password</label>
            <input name="password" />
            <button className="yellow-button">Sign In</button>
            <div className="checkbox-container">
              <input type="checkbox" />
              <p>Remember me</p>
            </div>
            <p>
              Forgot
              <Link to="/reset" className="link">
                &nbsp;username
              </Link>
              &nbsp;or
              <Link to="/reset" className="link">
                &nbsp;password
              </Link>
            </p>
          </form>
        </section>
        <section className="signup-section">
          <img />
          <h2>Don't have an account?</h2>
          <Link to="/dashboard" className="yellow-button">
            Sign Up
          </Link>
        </section>
      </div>

      // <div className="login-page">
      //   <div className="login-bg" />
      //   <Link to="/" className="header">
      //     <div className="globe">
      //       <FontAwesomeIcon icon="globe" />
      //     </div>
      //     <h1 className="app-name">Arsenalsssss</h1>
      //   </Link>
      //   <form
      //     className="login-form"
      //     onChange={this.onFieldChange}
      //     onSubmit={this.onSubmit}
      //   >
      //     {errors.global && <p className="global-errors">{errors.global}</p>}
      //     <div className="input-group">
      //       <label className="title">username</label>
      //       <input
      //         type="text"
      //         name="username"
      //         style={{
      //           border: !!errors.username ? '2px solid #e87c7c' : 'none'
      //         }}
      //       />
      //       {errors.username && (
      //         <p className="inline-errors">{errors.username}</p>
      //       )}
      //     </div>
      //     <div className="input-group">
      //       <label className="title">password</label>
      //       <input
      //         type="password"
      //         name="password"
      //         style={{
      //           border: !!errors.password ? '2px solid #e87c7c' : 'none'
      //         }}
      //       />
      //       {errors.password && (
      //         <p className="inline-errors">{errors.password}</p>
      //       )}
      //     </div>
      //     <div className="input-group">
      //       <div className="switch-page">
      //         <p className="text">Don't have an account?</p>
      //         <Link to="/signup" className="signup-link">
      //           &nbsp;Signup
      //         </Link>
      //       </div>
      //       <button type="submit" className="login-button button">
      //         Login
      //       </button>
      //     </div>
      //   </form>
      //   <div className="overlay" />
      //   <div className="bg" />
      // </div>
    )
  }
}

export default connect(
  null,
  { startLogin }
)(LoginPage)
