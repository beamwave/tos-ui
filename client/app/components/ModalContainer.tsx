import React from 'react'
import { connect } from 'react-redux'

// import all modal components
import AddImageModal from './modals/AddImageModal'
import gradeAdjustmentsModal from './modals/GradeAdjustmentsModal'

// import modal type constants
import {
  ADD_IMAGE_MODAL,
  GRADE_ADJUSTMENTS_MODAL
} from '../constants/modaltypes'

// modal directory based on props.modalType
const MODAL_COMPONENTS = {
  ADD_IMAGE_MODAL: AddImageModal,
  GRADE_ADJUSTMENTS_MODAL: gradeAdjustmentsModal
}

export const ModalContainer = props => {
  // if no modal set in store, do not render one
  if (!props.modalType) {
    return null
  }

  // object lookup
  const SpecificModal = MODAL_COMPONENTS[props.modalType]

  return <SpecificModal />
}

const mapStateToProps = state => ({
  modalType: state.modal.modalType
})

export default connect(mapStateToProps)(ModalContainer)
