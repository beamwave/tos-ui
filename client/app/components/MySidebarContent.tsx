import React, { Component } from 'react'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export class MySidebarContent extends Component {
  // @ts-ignore
  render = () => {
    return (
      <div className="sidebar-content">
        <div className="trash">
          <FontAwesomeIcon icon="trash-alt" />
        </div>
        <div className="categories">
          <div className="category">
            <p>Staging</p>
            <div className="angle-right">
              <FontAwesomeIcon icon="angle-right" />
            </div>
          </div>
          <div className="category">
            <p>Moodboards</p>
            <div className="angle-right">
              <FontAwesomeIcon icon="angle-right" />
            </div>
          </div>
          <div className="category">
            <p>Collections</p>{' '}
            <div className="angle-right">
              <FontAwesomeIcon icon="angle-right" />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default connect()(MySidebarContent)
