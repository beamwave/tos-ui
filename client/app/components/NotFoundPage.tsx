import React from 'react'

export const NotFoundPage = () => <div>Sorry, page not found.</div>

export default NotFoundPage
