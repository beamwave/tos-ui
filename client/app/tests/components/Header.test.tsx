import React, { Component } from 'react'
import { shallow } from 'enzyme'
import { Header } from '../../components/Header'

test('should render Header component', () => {
  const wrapper = shallow(
    <Header
      photo="url"
      username="mock-user"
      isAuthenticated={true}
      trainingWheels={false}
      startLogout={() => console.log('placeholder')}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
