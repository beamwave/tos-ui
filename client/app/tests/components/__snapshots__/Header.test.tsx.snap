// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<header
  className="nav-header"
>
  <div
    className="plus"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="plus"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <p
      className="hint"
    >
      upload image
    </p>
  </div>
  <div
    className="settings"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="cog"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <p
      className="hint"
    >
      settings
    </p>
  </div>
  <div
    className="protocol"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="unlock"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <p
      className="hint"
    >
      active
       training wheels protocol )}
    </p>
  </div>
  <div
    className="search-group"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="search"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
    <form
      className="submit"
      onSubmit={[Function]}
    >
      <input
        className="input spawnSearch"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <div
        className="type-choice"
      >
        <div
          className="angle-down"
        >
          <FontAwesomeIcon
            border={false}
            className="icon"
            fixedWidth={false}
            flip={null}
            icon="angle-down"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            transform={null}
          />
        </div>
        <select
          className="select teal"
          onChange={[Function]}
          value="art"
        >
          <option
            value="art"
          >
            art
          </option>
          <option
            value="comments"
          >
            comments
          </option>
          <option
            value="clothes"
          >
            clothes
          </option>
          <option
            value="people"
          >
            people
          </option>
          <option
            value="music"
          >
            music
          </option>
          <option
            value="quora"
          >
            quora
          </option>
          <option
            value="bookmarks"
          >
            bookmarks
          </option>
          <option
            value="recipes"
          >
            recipes
          </option>
          <option
            value="vocabulary"
          >
            vocabulary
          </option>
          <option
            value="notes"
          >
            notes
          </option>
          <option
            value="timeline"
          >
            timeline
          </option>
        </select>
      </div>
      <div
        className="search-options"
      >
        <p
          className="save-search"
        >
          save search
        </p>
        <p
          className="advanced-search"
        >
          advanced search
        </p>
      </div>
    </form>
  </div>
  <Dropdown
    active={false}
    addonType={false}
    className="dropdown-root"
    direction="down"
    inNavbar={false}
    isOpen={false}
    nav={false}
    setActiveFromChild={false}
    toggle={[Function]}
  >
    <DropdownToggle
      aria-haspopup={true}
      className="dropdown-toggle"
      color="secondary"
    >
      <p
        className="nav-username"
      >
        mock-user
      </p>
      <div
        className="image"
        style={
          Object {
            "background": "url(url) 50% 5% / 110% no-repeat",
            "marginRight": 15,
          }
        }
      />
      <FontAwesomeIcon
        border={false}
        className="icon fa-angle-down"
        fixedWidth={false}
        flip={null}
        icon="angle-down"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </DropdownToggle>
    <DropdownMenu
      className="dropdown-menu"
      flip={true}
      right={true}
      style={
        Object {
          "display": "none",
        }
      }
      tag="div"
    >
      <DropdownItem
        className="dropdown-item"
        tag="button"
        toggle={true}
      >
        <Link
          className="profile"
          replace={false}
          to="/profile"
        >
          Profile
        </Link>
      </DropdownItem>
      <DropdownItem
        className="dropdown-item"
        onClick={[Function]}
        tag="button"
        toggle={true}
      >
        <p
          className="logout"
        >
          Logout
        </p>
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
</header>
`;
